### ISSUE TITLE
Run Alertmanager with -target=all (monolithic mode)

### ISSUE THREAD

**Issue by pracucci** [grafana/mimir#2773]
Alertmanager is the only Mimir component (among the ones used in a standard setup) that isn't started in monolithic mode (`-target=all`.

**Question**: what if we start Alertmanager with `-target=all`?

**Comment by pracucci** [comment]
I'm currently working on a the experimental read-write deployment mode (https://github.com/grafana/mimir/issues/2749) and want to start Alertmanager by default. Adding Alertmanager to monolithic mode will make it consistent.

**Comment by jedd** [comment]
Actually, that's not quite true : )

```
/ # /bin/mimir -modules
alertmanager
all
compactor *
distributor *
flusher
ingester *
overrides-exporter
purger *
querier *
query-frontend *
query-scheduler
ruler *
store-gateway *

Modules marked with * are included in target All.

```

Would it be worth qualifying that for new adopters at:  https://grafana.com/docs/mimir/latest/operators-guide/architecture/components/alertmanager/

**Comment by pracucci** [comment]
Yeah, I consider `overrides-exporter` a really optional one. About `query-scheduler` we currently can't include it in monolithic mode (we need to guarantee only 2 replicas of it running, not more than that). We'll do some work around it for the read-write deployment, so after that we may consider to add it to monolithic mode.

Following this reasoning, `alertmanager` is the only one left.

**Comment by wilfriedroset** [comment]
If users run a single monolithic mimir (meaning a single alertmanager) they will face the bug explained in #2910.
Starting from two instances, until the bug is fixed, they should be ok.
